### PR TITLE
Enhanced find with appointment searching and search field composition

### LINKED ISSUE #100: Enhance Find command to search through Appointments
This enhancement will be heavily affected by #96 - it would be better for that issue to be resolved first before starting on this one.

### PR DESCRIPTION
**Pull request by WZWren**
Resolves #100 on merge.

Additionally, `find` can now compose different search conditions together. This will be resolved as an `AND` operation between the various search fields.

### CODE REVIEW

**Comment by hyc17003** [comment]
Very useful enhancements to Find function, this LGTM.

File: src/main/java/seedu/address/model/person/Appointment.java [hunk 1]
```diff
@@ -144,6 +144,25 @@ public static boolean isValidAppointmentDelimit(String test, InputSource origin)
         }
     }
 
+    /**
+     * Returns true if the period represented by this {@code Appointment} overlaps with the period
+     * of {@code target} on the same day, inclusive of the {@code start} and {@code end}.<br>
+     *
+     * This method defines an incompatibility relationship between 2 {@code Appointment} objects.
+     */
+    public boolean overlaps(Appointment target) {
```
Comment (kanna-1) [inline src/main/java/seedu/address/model/person/Appointment.java:153]: Should this method be renamed to sound more like a Boolean instead as per the Java Coding Standard?

**Comment by kanna-1** [comment]
Other than the potential Java Coding Standard violation, this LGTM.

File: src/main/java/seedu/address/model/person/Appointment.java (hunk 1 above)
Comment (WZWren) [reply src/main/java/seedu/address/model/person/Appointment.java]: This is true - I will rename the method to a more appropriate `hasOverlap` instead.

**Comment by WZWren** [comment]
Thank you for reviewing the code @hyc17003 @kanna-1, I will be merging this now.

**Pull request merged**